pylearn2/datasets/dense_design_matrix.py

```diff
@@ -264,13 +264,14 @@ def _check_labels(self):
             assert self.y.ndim <= 2
             assert np.all(self.y < self.y_labels)
 
-    @functools.wraps(Dataset.iterator)
+    # Cannot wrap the iterator method cause it is abstract
+    # @functools.wraps(Dataset.iterator)
```

Comment:
Updating `@pylearn2.utils.wraps` may be the way to go in the long run, or find a way to make NumPyDoc work with an abstract method.
But for the moment, let's do one thing at a time. First, let's merge the new version of HDF5Dataset, with inherited docstrings that work and show up in the online doc. Then we can impose an interface on Dataset and make sure it works nicely with docstrings, tests, and online doc.